packages/vite/src/node/plugins/resolve.ts

```diff
@@ -77,168 +77,177 @@ export function resolvePlugin(baseOptions: InternalResolveOptions): Plugin {
   }
   let server: ViteDevServer | undefined
 
-  return {
-    name: 'vite:resolve',
+  const resolveId: Plugin['resolveId'] = (id, importer, resolveOpts, ssr) => {
+    const targetWeb = !ssr || ssrTarget === 'webworker'
+
+    // this is passed by @rollup/plugin-commonjs
+    const isRequire =
+      resolveOpts &&
+      resolveOpts.custom &&
+      resolveOpts.custom['node-resolve'] &&
+      resolveOpts.custom['node-resolve'].isRequire
+
+    const options = isRequire ? requireOptions : baseOptions
+
+    let res
```

Comment:
You should type this so we don't have floating typeless variables

Reply:
Not my changes :P

Reply:
Not my changes as well